remoteconfig/client.go

```diff
@@ -0,0 +1,223 @@
+	opts = append(opts, internal.WithQueryParam("validateOnly", strconv.FormatBool(validateOnly)))
+
+	var data RemoteConfig
+
+	resp, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodPut,
+			URL:    c.getRootURL(),
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &Response{
+		RemoteConfig: data,
+		Etag:         resp.Header.Get("Etag"),
+	}
+
+	return result, nil
+}
+
+// ListVersions https://firebase.google.com/docs/reference/remote-config/rest/v1/projects.remoteConfig/listVersions
+func (c *Client) ListVersions(options *ListVersionsOptions) (*ListVersionsResponse, error) {
+	var opts []internal.HTTPOption
+	if options.PageSize != 0 {
+		opts = append(opts, internal.WithQueryParam("pageSize", strconv.FormatInt(options.PageSize, 10)))
+	}
+
+	if options.PageToken != "" {
+		opts = append(opts, internal.WithQueryParam("pageToken", options.PageToken))
+	}
+
+	if options.EndVersionNumber != "" {
+		opts = append(opts, internal.WithQueryParam("endVersionNumber", options.EndVersionNumber))
+	}
+
+	if options.StartTime != "" {
+		opts = append(opts, internal.WithQueryParam("startTime", options.StartTime))
+	}
+
+	if options.EndTime != "" {
+		opts = append(opts, internal.WithQueryParam("endTime", options.EndTime))
+	}
+
+	var data ListVersionsResponse
+	url := fmt.Sprintf("%s:listVersions", c.getRootURL())
+
+	_, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodGet,
+			URL:    url,
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &data, nil
+}
+
+// Rollback https://firebase.google.com/docs/reference/remote-config/rest/v1/projects.remoteConfig/rollback
+func (c *Client) Rollback(versionNumber string) (*Template, error) {
+	if versionNumber == "" {
+		return nil, errors.New("versionNumber is required to rollback a Remote Config template")
+	}
+
+	var data Template
+	url := fmt.Sprintf("%s:listVersions", c.getRootURL())
+
+	_, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodPost,
+			Body: internal.NewJSONEntity(
+				struct {
+					VersionNumber string `json:"versionNumber"`
+				}{
+					VersionNumber: versionNumber,
+				},
+			),
+			URL: url,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &data, nil
+}
+
+// NewClient returns the default remote config
+func NewClient(ctx context.Context, c *internal.RemoteConfig) (*Client, error) {
+	if c.ProjectID == "" {
+		return nil, errors.New("project ID is required to access Firebase Cloud Remote Config client")
+	}
+
+	hc, _, err := internal.NewHTTPClient(ctx, c.Opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		hc:        hc,
+		projectID: c.ProjectID,
+	}, nil
+}
+
+// GetTemplate will retrieve the latest template
+func (c *Client) GetTemplate(ctx context.Context) (*Template, error) {
```

Comment:
Please see my comments above for `GetTemplate` functions. You can implement one if you don't plan to implement other versions.... please remove any placeholder functions.